--- bluez-utils-2.7/acinclude.m4.orig	2004-05-09 12:39:43.000000000 +0200
+++ bluez-utils-2.7/acinclude.m4	2004-05-21 00:48:24.694028984 +0200
@@ -153,7 +153,7 @@
 	if test "$cups_enable" = "yes"; then
 		AC_MSG_CHECKING(for CUPS backend directory)
 
-		if (test -d "$cups_prefix/lib/cups/backend"); then
+		if true; then
 			CUPS_BACKEND_DIR="$cups_prefix/lib/cups/backend"
 		else
 			cups_enable=no
